client/src/ActionsAndReducers/dbWargames/wargames_Reducer.ts

```diff
@@ -22,107 +32,121 @@ const initialState = {
-      break
-
-    case ActionConstant.ADD_NEW_CHANNEL:
+  const handler = handlers[action.type]
 
-      newChannel = channelTemplate
-      newChannel.name = action.payload.name
-      newChannel.uniqid = action.payload.uniqid
-      newState.data[tab].channels.push(newChannel)
-      break
-
-    case ActionConstant.SET_SELECTED_CHANNEL:
-      newState.data[tab].selectedChannel = action.payload
-      break
-
-    case ActionConstant.DELETE_SELECTED_CHANNEL:
+  if (handler) {
+    handler(newState, action, tab)
+  }
 
-      channelIndex = newState.data[tab].channels.findIndex((channel: ChannelTypes) => channel.name === action.payload.name)
-      newState.data[tab].channels.splice(channelIndex, 1)
-      newState.data[tab].selectedChannel = ''
-      break
+  switch (action.type) {
+    // case ActionConstant.ALL_WARGAME_NAMES_SAVED:
+    //   handleAllWargameNamesSaved(newState, action.payload)
+    //   break
+
+    // case ActionConstant.SET_CURRENT_WARGAME:
+    //   handleSetCurrentWargame(newState, action.payload)
+    //   break
+    // case ActionConstant.SET_EXPORT_WARGAME:
+    //   handleSetExportWargame(newState, action.payload)
+    //   break
+    // case ActionConstant.SET_CURRENT_GAME_SETUP_TAB:
+    //   newState.currentTab = action.payload
+    //   break
+
+    // case ActionConstant.SET_GAME_SETUP_DATA:
+    //   newState.data[tab] = { ...newState.data[tab], ...action.payload }
+    //   break
+
+    // case ActionConstant.SET_TAB_SAVED:
+    //   newState.data[tab].dirty = false
+    //   break
+
+    // case ActionConstant.SET_TAB_UNSAVED:
+    //   newState.data[tab].dirty = true
+    //   break
+
+    // case ActionConstant.ADD_NEW_FORCE:
+
+    //   newForce = forceTemplate
+    //   newForce.name = action.payload.name
+    //   newForce.uniqid = action.payload.uniqid
+
+    //   newState.data[tab].forces.unshift(newForce)
+    //   break
+
+    // case ActionConstant.SET_FORCE_COLOR:
+    //   newState.data[tab].forces.find((force: ForceData) => force.name === newState.data[tab].selectedForce.name).color = action.payload
+    //   break
+
+    // case ActionConstant.SET_SELECTED_FORCE:
+    //   newState.data.forces.selectedForce = action.payload as any
+    //   break
+
+    // case ActionConstant.ADD_NEW_CHANNEL:
+    //   console.log('newChannel')
+    //   newChannel = channelTemplate
+    //   newChannel.name = action.payload.name
+    //   newChannel.uniqid = action.payload.uniqid
+    //   newState.data[tab].channels.push(newChannel)
+    //   break
+
+    // case ActionConstant.SET_SELECTED_CHANNEL:
+    //   newState.data[tab].selectedChannel = action.payload
+    //   break
+
+    // case ActionConstant.DELETE_SELECTED_CHANNEL:
+
+    //   channelIndex = newState.data[tab].channels.findIndex((channel: ChannelTypes) => channel.name === action.payload.name)
+    //   newState.data[tab].channels.splice(channelIndex, 1)
+    //   newState.data[tab].selectedChannel = ''
+    //   break
 
```

Comment:
should the next block be in the `handlers` @lilitkarapetyan ?

Reply:
yes